core/src/main/scala/com/criteo/cuttle/Executor.scala

```diff
@@ -818,15 +853,17 @@ class Executor[S <: Scheduling] private[cuttle] (val platforms: Seq[ExecutionPla
                   unsafeDoRun(execution, promise)
                 case Paused =>
                   execution.streams.debug(s"Delayed because job ${execution.job.id} is paused")
+                  // we attach this callback to freshly created "Paused" execution
                   execution.onCancel { () =>
                     val cancelNow = atomic { implicit tx =>
-                      val isStillPaused = pausedState.get(job.id).getOrElse(Map.empty).contains(execution)
-                      if (isStillPaused) {
-                        val pausedExecutions = pausedState.get(job.id).getOrElse(Map.empty).filterKeys(_ == execution)
-                        pausedState += (job.id -> pausedExecutions)
+                      val job2Execution = pausedState.get(job.id).collectFirst {
```

Comment:
Please add a comment, and possibly rename job2Execution to something a bit more self-explanatory

Reply:
done